extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php

```diff
@@ -0,0 +1,243 @@
+<?php
+
+class CommunityPageSpecialController extends WikiaSpecialPageController {
+	const DEFAULT_TEMPLATE_ENGINE = \WikiaResponse::TEMPLATE_ENGINE_MUSTACHE;
+	protected $usersModel;
+	protected $wikiModel;
+
+	public function __construct() {
+		parent::__construct( 'Community' );
+		$this->usersModel = new CommunityPageSpecialUsersModel();
+		$this->wikiModel = new CommunityPageSpecialWikiModel();
+	}
+
+	public function index() {
+		$this->specialPage->setHeaders();
+		$this->getOutput()->setPageTitle( $this->msg( 'communitypage-title' )->plain() );
+		$this->addAssets();
+
+		$this->wg->SuppressPageHeader = true;
+		$this->wg->SuppressWikiHeader = true;
+		$this->wg->SuppressFooter = true;
+
+		// queue i18n messages for export to JS
+		JSMessages::enqueuePackage( 'CommunityPageSpecial', JSMessages::EXTERNAL );
+
+		// remove user styles and js
+		$this->getOutput()->disallowUserJs();
+
+		$this->response->setValues( [
+			'adminWelcomeMsg' => $this->msg( 'communitypage-tasks-admin-welcome' )->text(),
+			'pageListEmptyText' => $this->msg( 'communitypage-page-list-empty' )->plain(),
+			'showPopupMessage' => true,
+			'popupMessageText' => 'This is just a test message for the popup message box',
+			'userIsMember' => CommunityPageSpecialHelper::userHasEdited( $this->wg->User ),
+			'pageTitle' => $this->msg( 'communitypage-title' )->plain(),
+			'topContributors' => $this->sendRequest( 'CommunityPageSpecialController', 'getTopContributorsData' )
+				->getData(),
+			'topAdmins' => $this->sendRequest( 'CommunityPageSpecialController', 'getTopAdminsData' )
+				->getData(),
+			'recentlyJoined' => $this->sendRequest( 'CommunityPageSpecialController', 'getRecentlyJoinedData' )
+				->getData(),
+			'recentActivityModule' => $this->getRecentActivityData(),
+		] );
+	}
+
+	public function header() {
+		$isMember = CommunityPageSpecialHelper::userHasEdited( $this->wg->User );
+
+		$this->response->setValues( [
+			'inviteFriendsText' => $this->msg( 'communitypage-invite-friends' )->plain(),
+			'headerWelcomeMsg' => $this->msg( 'communitypage-tasks-header-welcome' )->plain(),
+			'pageListEditText' => $this->msg( 'communitypage-page-list-edit' )->plain(),
+			'thisMonthText' => $this->msg( 'communitypage-this-month' )->plain(),
+			'showMonthlySummary' => $isMember,
+			'showAdminsSummary' => !$isMember,
+			'statPagesTitle' => $this->msg( 'communitypage-pages' )->plain(),
+			'statPagesNumber' => 45,
+			'statPageViewsTitle' => $this->msg( 'communitypage-pageviews' )->plain(),
+			'statPageViewsNumber' => '1,049',
```

Comment:
This was for demonstration purposes: It is the stuff that we haven't hooked up to real data yet.